kubernetes/terraform/modules/kubernetes/kubernetes_dashboard.tf

```diff
@@ -0,0 +1,309 @@
+    resource_names = ["heapster", "dashboard-metrics-scraper"]
+  }
+  rule {
+    verbs          = ["get"]
+    api_groups     = [""]
+    resources      = ["services/proxy"]
+    resource_names = ["heapster", "http:heapster:", "https:heapster:", "dashboard-metrics-scraper", "http:dashboard-metrics-scraper"]
+  }
+}
+
+resource "kubernetes_cluster_role" "kubernetes_dashboard" {
+  metadata {
+    name   = "kubernetes-dashboard"
+    labels = { k8s-app = "kubernetes-dashboard" }
+  }
+  rule {
+    verbs      = ["get", "list", "watch"]
+    api_groups = ["metrics.k8s.io"]
+    resources  = ["pods", "nodes"]
+  }
+}
+
+resource "kubernetes_role_binding" "kubernetes_dashboard" {
+  metadata {
+    name      = "kubernetes-dashboard"
+    namespace = "kubernetes-dashboard"
+    labels    = { k8s-app = "kubernetes-dashboard" }
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = "kubernetes-dashboard"
+    namespace = "kubernetes-dashboard"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "Role"
+    name      = "kubernetes-dashboard"
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "kubernetes_dashboard" {
+  metadata {
+    name = "kubernetes-dashboard"
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = "kubernetes-dashboard"
+    namespace = "kubernetes-dashboard"
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "kubernetes-dashboard"
+  }
+}
+
+resource "kubernetes_deployment" "kubernetes_dashboard" {
+  metadata {
+    name      = "kubernetes-dashboard"
+    namespace = "kubernetes-dashboard"
+    labels    = { k8s-app = "kubernetes-dashboard" }
+  }
+  spec {
+    replicas = 1
+    selector {
+      match_labels = { k8s-app = "kubernetes-dashboard" }
+    }
+    template {
+      metadata {
+        labels = { k8s-app = "kubernetes-dashboard" }
+      }
+      spec {
+        volume {
+          name = "kubernetes-dashboard-certs"
+          secret {
+            secret_name = "kubernetes-dashboard-certs"
+          }
+        }
+        volume {
+          name = "tmp-volume"
+        }
+        container {
+          name  = "kubernetes-dashboard"
+          image = "kubernetesui/dashboard:v2.0.0-rc7"
+          args  = ["--auto-generate-certificates", "--namespace=kubernetes-dashboard"]
+          port {
+            container_port = 8443
+            protocol       = "TCP"
+          }
+          volume_mount {
+            name       = "kubernetes-dashboard-certs"
+            mount_path = "/certs"
+          }
+          volume_mount {
+            name       = "tmp-volume"
+            mount_path = "/tmp"
+          }
+          liveness_probe {
+            http_get {
+              path   = "/"
+              port   = "8443"
+              scheme = "HTTPS"
+            }
+            initial_delay_seconds = 30
+            timeout_seconds       = 30
+          }
+          image_pull_policy = "Always"
```

Comment:
Given the image is pinned to tag `v2.0.0-rc7` we can set the `image_pull_policy` to `IfNotPesent` as it is unnecessary to re-pull the same image.